### PR TITLE
MS Exchange - Add missing ECS mappings & ability to configure request timeout

### LINKED ISSUE #5291: [Microsoft Exchange Online Message Trace] Configurable `request.timeout`
The current integration has a hard coded timeout of 60s:

https://github.com/elastic/integrations/blob/cbe1cc8ec3164ff3ff1c599ca23a8947e2a7ab9f/packages/microsoft_exchange_online_message_trace/data_stream/log/agent/stream/httpjson.yml.hbs#L13

The issue here is that if you set a large `initial_interval` value, because you want to do something like backfill all available data. The integration will always timeout before it gets a response from the API. To be able to successfully set large `initial_interval` values, you'll need to be able to configure the `request.timeout` value to a longer duration.

### PR DESCRIPTION
**Pull request by BenB196**
Enhancement & Bug

## What does this PR do?

Two changes were made:

1. Added missing `event.start` and `event.created` ECS mappings
2. Added the ability to configure the `request.timeout` field

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

I made this a single PR and only bumped by patch as both of these things are relatively small changes. I can split into 2 different PRs if desired.

## How to test this PR locally

1. Spin up stack install old version of package
2. Update to new version of package
3. Observe that the missing field mappings are now present in the component template
4. Observe that you can now configure the request.timeout value. Confirm that the default value is 60s and that it is set when adding the integration to a policy. Confirm that changing the timeout value also changes it in the policy.

## Related issues

- Closes #5291

### CODE REVIEW

File: packages/microsoft_exchange_online_message_trace/changelog.yml [hunk 1]
```diff
@@ -1,3 +1,11 @@
+- version: "0.3.1"
```
Comment (efd6) [inline packages/microsoft_exchange_online_message_trace/changelog.yml:1]: By convention we bump minor for version-compatible enhancements.

Comment (BenB196) [reply packages/microsoft_exchange_online_message_trace/changelog.yml]: Updated from 0.3.1 to 0.4.0

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**